apps/portal/src/hooks/use-search.ts

```diff
@@ -0,0 +1,116 @@
+'use client';
+//
+import { useState, useEffect, useMemo } from 'react';
+import useSWR from 'swr';
+import { EventSummary, TeamSummary } from '@lems/types/api/portal';
+
+export interface SearchResult {
+  type: 'team' | 'event';
+  id: string;
+  data: TeamSummary | EventSummary;
+}
+
+export interface SearchResponse {
+  results: SearchResult[];
+  total: number;
+  query: string;
+  type: string;
+  status: string;
+}
+
+export interface UseSearchOptions {
+  minQueryLength?: number;
+  debounceMs?: number;
+}
+
+export const useSearch = (options: UseSearchOptions = {}) => {
+  const { minQueryLength = 2, debounceMs = 300 } = options;
+  
+  const [query, setQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState('');
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setDebouncedQuery(query);
+    }, debounceMs);
+
+    return () => clearTimeout(timer);
+  }, [query, debounceMs]);
+
+  const searchUrl = useMemo(() => {
+    if (!debouncedQuery || debouncedQuery.length < minQueryLength) {
+      return null;
+    }
+
+    const params = new URLSearchParams({
+      q: debouncedQuery,
+      type: 'all',
+      limit: '50'
+    });
+
+    return `/portal/search?${params.toString()}`;
+  }, [debouncedQuery, minQueryLength]);
+
+  const { data: searchResponse, error, isLoading } = useSWR<SearchResponse>(
+    searchUrl,
+    { 
+      revalidateOnFocus: false,
+      revalidateOnReconnect: false,
+      dedupingInterval: 1000
+    }
+  );
+
+  const processedResults = useMemo(() => {
+    if (!searchResponse?.results) return [];
+
+    return searchResponse.results.map(result => ({
```

Comment:
This is very very complex. Why not just return the data properly from the backend? We're doing expensive mapping onn both back and front for no reason. This is a purpose built endpoint anyway